### ISSUE TITLE
why unsuport on windows

### ISSUE THREAD

**Issue by GhengS** [rvaiya/warpd#117]
**What an amazing project!**
It would be nice if it could also be used under Windows
Is there a way to do it please?

**Comment by GhengS** [comment]
作为一位以windows为主使用平台的用户，
现在我的内心充满了羡慕，嫉妒和期待

**Comment by GhengS** [comment]
As a Windows-based user, I am now filled with envy, jealousy and anticipation.
Now my heart is full of envy, jealousy and expectation

**Comment by GhengS** [comment]
As a Windows-based user, I am now filled with envy, jealousy and anticipation.

**Comment by rvaiya** [comment]
I may get around to a windows port at some point in the future, but I already have a lot in my backlog. Patches are welcome :).

**Comment by rensx5514** [comment]
Wish it! It's so much fun coming to windows!

**Comment by yyy33** [comment]
> As a Windows-based user, I am now filled with envy, jealousy and anticipation.

windows has autohotkey

**Comment by slacksystem** [comment]
> > As a Windows-based user, I am now filled with envy, jealousy and anticipation.
> 
> windows has autohotkey

Do you know of any autohotkey scripts that work like this? AHK is only as good as the scripts you have

**Comment by nerasw** [comment]
> > > > As a Windows-based user, I am now filled with envy, jealousy and anticipation.
> > > 
> > > 
> > > windows has autohotkey
> > 
> > 
> > Do you know of any autohotkey scripts that work like this? AHK is only as good as the scripts you have
> 
> there's this one: https://github.com/EsportToys/TPMouse

i personally like this one: https://github.com/wirekang/mouseable
TPMouse has bad customization, but is what really make it worse than mouseable is that it doesn't block u from typing upon call. It does not remove focus from current window and keep typing while moving cursor. At least for me (tested in MVS and couple browsers).

**Comment by slacksystem** [comment]
> > > > > > As a Windows-based user, I am now filled with envy, jealousy and anticipation.
> > > > > 
> > > > > 
> > > > > windows has autohotkey
> > > > 
> > > > 
> > > > Do you know of any autohotkey scripts that work like this? AHK is only as good as the scripts you have
> > > 
> > > 
> > > there's this one: https://github.com/EsportToys/TPMouse
> > 
> > 
> > i personally like this one: https://github.com/wirekang/mouseable TPMouse has bad customization, but is what really make it worse than mouseable is that TPM doesn't block u from typing upon call. It does not remove focus from current window and keep typing while moving cursor. At least for me (tested in MVS and couple browsers).
> 
> Just tested, moving the cursor works fine without any stray key inputs. Did you perhaps press the wrong keys?
> 
> I personally like it for the way normal mode works, it feels much more natural than the fixed-speed cursor motion of the others, reminds me of playing Super Mario.

Neither are autohotkey scripts, which is what was suggested by @yyy33, but TPMouse looks interesting I might try it. Haven't done anything with basic in decades, but it'll be good to get back to the basics (sorry for the pun). I don't like t he idea of having to run chrome or edge just to use the software, which unless I'm misunderstanding is what's required for mouseable

**Comment by rvaiya** [comment]
There is in fact a beta windows version sitting in master. I pushed it some time ago, but haven't had time to make a proper release. I don't have time to test it at the moment (or remember what state I left it in), but I just published the binary built from the tip of master [here](https://github.com/rvaiya/warpd/releases/download/v1.3.6-beta-windows/warpd.exe) if you want to give it a try. YMMV

**Comment by Avastgard** [comment]
Will this beta release work with Windows 11?

**Comment by slacksystem** [comment]
> There is in fact a beta windows version sitting in master. I pushed it some time ago, but haven't had time to make a proper release. I don't have time to test it at the moment (or remember what state I left it in), but I just published the binary built from the tip of master [here](https://github.com/rvaiya/warpd/releases/download/v1.3.6-beta-windows/warpd.exe) if you want to give it a try. YMMV

I'm confused, are you saying this Vimux is a beta windows version of warpd?

**Comment by ChizhovLeonid** [comment]
Yep, I got the same error. But once it turned out and the program remained in the tray. But I couldn't get into the config settings.

**Comment by crestofthebeast** [comment]
A more verbose report:

When i boot warpd on my Win11 machine, I receive the same error as Avastgard and Chizhov.
I get an error popup, and a black rectangle that I assume is some sort of unrendered GUI?
- Selecting abort closes the application.

- Selecting retry puts the application in my try, but kills it shortly after.

- Selecting abort results in a second, very similar error:
```
---------------------------
Microsoft Visual C++ Runtime Library
---------------------------
Assertion failed!

Program: ...
File: src/platform/windows/filemon.c
Line: 31

Expression: fh != INVALID_HANDLE_VALUE

For information on how your program can cause an assertion
failure, see the Visual C++ documentation on asserts

(Press Retry to debug the application - JIT must be enabled!)
---------------------------
Abort      Retry      Ignore
---------------------------
```

After clicking ignore again, similar behaviour to initially selecting 'Retry' is observed.

**Comment by crestofthebeast** [comment]
After poking around at `/platform/windows/filemon.c`, it seems like the issue is with file creation. Booting in admin mode doesn't help the problem, but perhaps it's a directory access issue?

**Comment by felixhudson** [comment]
Hey, I tried to fix this and failed, so I implemented my own in csharp.

https://github.com/felixhudson/cwarpd

Hope you find this useful.

@GhengS enjoy!

**Comment by rvaiya** [comment]
Apologies. The original build I released was broken. I have fixed the issues and released another build which can be found here:

https://github.com/rvaiya/warpd/releases/tag/v1.3.7-windows-alpha (read the release notes).

Please be aware that this is currently alpha quality software. I don't have much time to maintain the windows version, and don't dogfood it (so bugs are less likely to be discovered). That said, feel free to post issues as they arise.

**Comment by aafutin** [comment]
> 

doesn't work on windows for me. just black window

**Comment by aafutin** [comment]
> Apologies. The original build I released was broken. I have fixed the issues and released another build which can be found here:
> 
> https://github.com/rvaiya/warpd/releases/tag/v1.3.7-windows-alpha (read the release notes).
> 
> Please be aware that this is currently alpha quality software. I don't have much time to maintain the windows version, and don't dogfood it (so bugs are less likely to be discovered). That said, feel free to post issues as they arise.

doesn't work on windows for me. just black window

**Comment by rgkirch** [comment]
Like @Salted-Bread I can't use it on a second screen. If I mouse over my second screen and initiate hint mode then I get this:
<img width="310" alt="image" src="https://github.com/rvaiya/warpd/assets/6143833/98391498-0926-4aab-8127-e64a75abbfb6">
I think you can get away with closing this ticket though since it does work on windows. Bugs can be their own tickets...

**Comment by Equilibrium94** [comment]
> this one: https://github.com/wirekang/mouseable
> TPMouse has bad customization, but is what really make it worse than mouseable is that TPM doesn't block u from typing upon call. It does

Can you explain how mouseable works? I cant figure out how that guy is controlling the mouse. it seems worse than warpd?

**Comment by Frestein** [comment]
You can try [jwno](https://github.com/agent-kilo/jwno). It supports interaction with GUIs using your keyboard.
https://agent-kilo.github.io/jwno/features.html

**Comment by Frestein** [comment]
Also, the fork maybe can support Windows.
https://github.com/atuan26/warpd
